flexmeasures/data/services/sensors.py

```diff
@@ -56,34 +57,53 @@ def get_sensors(
     return db.session.scalars(sensor_query).all()
 
 
-def _get_sensor_bdf(sensor: Sensor, staleness_search: dict) -> BeliefsDataFrame | None:
-    """Get bdf for a given sensor with given search parameters."""
+def _get_sensor_bdfs_by_source(
+    sensor: Sensor, staleness_search: dict
+) -> DataFrameGroupBy | None:
+    """Get bdf split by source for a given sensor with given search parameters."""
     bdf = TimedBelief.search(
         sensors=sensor,
         most_recent_events_only=True,
         **staleness_search,
     )
     if bdf.empty:
         return None
-    return bdf
+    return bdf.groupby(level=["source"], group_keys=False)
```

Comment:
It should be possible to avoid the `DataFrameGroupBy` type annotation and thereby the extra dependency.
```suggestion
    return dict(tuple(bdf.groupby(level=["source"], group_keys=False)))
```